src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -260,9 +260,11 @@ private void ApplyBindingInformation(BindingConfiguration bindingConfig)
             Debug.Assert(this.ActiveSection?.RefreshCommand != null, "Refresh command is not set");
             // Run the refresh workflow, passing the connection information
             var refreshCmd = this.ActiveSection.RefreshCommand;
-            if (refreshCmd.CanExecute(bindingConfig))
+
+            var connection = bindingConfig.Project.CreateConnectionInformation();
+            if (refreshCmd.CanExecute(connection))
             {
-                refreshCmd.Execute(bindingConfig); // start the workflow
+                refreshCmd.Execute(connection); // start the workflow
```

Comment:
Miscellaneous: fixing a bug I introduced in a previous commit. The _refresh_ command is not strongly-typed, and I incorrectly changed the code to pass in the wrong type of argument. This wasn't picked up by existing tests, and doesn't cause an error at runtime (the command does a safe cast and does nothing if the supplied argument is not of the expected typed). I only noticed this when manually testing to check that everything in the legacy connected mode still works.
Rather than adding a test for this, I'll change the _refresh_ command to be strongly-typed too.